### ISSUE TITLE
Explore using AOT compilation with TruffleRuby to speed up fastlane

### ISSUE THREAD

**Issue by vikeri** [fastlane/fastlane#14386]
#### Motivation Behind Feature
tl;dr: Increase fastlane speed

While ruby is a great language, speed is not what it is optimized for. For bigger fastlane projects this means fastlane tasks can take a lot of time. One use cases of GraalVM is to increase cold start and performance with the drawback that it needs a compilation step.

#### Feature Description

1. Make fastlane run on [TruffleRuby](https://github.com/oracle/truffleruby) 
2. Add config that allows ahead-of-time compiling my fastlane setup in order to run everything faster.

Ideally when I edit my fastlane file and test it I run it as I do now, without AOT-compilation. This will maintain iteration speed when editing my Fastfile. But as soon as I'm happy with the setup I compile it and every execution after that is faster.

**Open questions**
This is a very experimental issue and I personally don't have a lot of experience with TruffleRuby hence:
- Is it even feasible to compile fastlane with TruffleRuby?
- If one gets it to compile, will it be faster than current implementation?

**Comment by vikeri** [comment]
Since this is a feature request `fastlane env` is not required.

**Comment by janpio** [comment]
Is there any evidence that the actual ruby part of fastlane is actually the "slow" part you are observing? Fastlane, by its nature, does a lot of shelling out to other tools and commands which usually take the most time.

**Comment by vikeri** [comment]
@janpio I'm not versed enough in the internals of fastlane so you might very well be right.
